msal/application.py

```diff
@@ -306,6 +309,8 @@ def get_authorization_request_url(
             nonce=None,
             domain_hint=None,  # type: Optional[str]
             claims_challenge=None,
+            code_challenge_method=None,
+            code_challenge=None,
```

Comment:
`acquire_token_interactive` and PKCE are two different features. I would suggest we separate them into 2 different PRs, so that their code review, subsequent wikis/samples (if any), and even future conversation reference ... would all be self-contained.

We may then choose to get the PKCE PR finished first.

Reply:
I think PKCE can be considered as a sub feature of acquire_token_interactive since PKCE as a whole involves two steps, one is generating `code_verifier` and `code_challenge`  and then sending `code_challenge` in step 1 to /oauth2/authorize request and the sending `code_verifier` in part 2 which is acquire_token_by_authorization code. But if we want to separate it into 2 PRs for the ease of code review, I can separate PKCE API support into a different PR 

However utilizing this PKCE feature(generating the code_verifier and code_challenge and then sending it in two steps) will still be in this PR. The actual support to be able to pass such parameters to the functions will be added once the PKCE PR is merged. 

PS: To be honest these changes were made to also get an E2E test working with the code_verifier and code_challenge being actually sent, how does it look like when a wrong code_verifier is sent, what error is encountered etc

Reply:
Your "PS" part is exactly the reason why the PKCE part of implementation, test, review, etc. should be in its own PR. Once that part is done, this acquire_token_interactive() can simply use it, without needing to test it.

Once you open a PKCE PR on the existing auth code flow, we will be get that reviewed and merged quickly.